pkgs/build-support/cc-wrapper/add-hardening.sh

```diff
@@ -1,32 +1,42 @@
-hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)
-# Intentionally word-split in case 'hardeningEnable' is defined in Nix.
-hardeningFlags+=(${hardeningEnable[@]})
+allHardeningFlags=(fortify stackprotector pie pic strictoverflow format relro bindnow)
 hardeningCFlags=()
 hardeningLDFlags=()
 
-declare -A hardeningDisableMap
+declare -A hardeningDisableMap=()
+declare -A hardeningEnableMap=()
 
-# Intentionally word-split in case 'hardeningDisable' is defined in Nix. The
+# Create table of unsupported flags for this toolchain.
+for flag in @hardening_unsupported_flags@; do
+  hardeningDisableMap[$flag]=1
+done
+
+# Intentionally word-split in case 'NIX_HARDENING_ENABLE' is defined in Nix. The
 # array expansion also prevents undefined variables from causing trouble with
 # `set -u`.
-for flag in ${hardeningDisable[@]} @hardening_unsupported_flags@
-do
-  hardeningDisableMap[$flag]=1
+for flag in ${NIX_HARDENING_ENABLE-}; do
```

Comment:
Could we do `NIX_CC_WRAPPER_@infixSalt@_HARDENING_ENABLE`? For the sake of clean laying I've wanted to use the `NIX_CC_WRAPPER` prefix instead, so might as well start with a new environment variable. `add-flags.sh` can take care of dealing with the `@infixSalt@` part; it should be fairly easy to cargo cult.

Reply:
Will do.

Reply:
@Ericson2314 This still needs to be changed to `NIX_@infixSalt@_HARDENING_ENABLE` (under the assumption that we'll stick to the old name for now, rather than `NIX_CC_WRAPPER_HARDENING_ENABLE`), right?

I must confess that I haven't found the time to learn how your cross compiling changes work, though I hope to soon-ish. Of course "soon" for me probably translates to "in a couple eons" for most people, given my schedule as of late...